lib/features/auth/presentation/pages/login_screen.dart

```diff
@@ -0,0 +1,159 @@
+        elevation: 0,
+        backgroundColor: Colors.white,
+      ),
+      body: BlocProvider(
+        create: (context) => getIt<LoginCubit>(),
+        child:
+            BlocConsumer<LoginCubit, LoginStates>(listener: (context, state) {
+          if (state is LoginLoadedState) {
+            Navigator.pushReplacement(context,
+                MaterialPageRoute(builder: (context) => const HomeScreen()));
+          }
+          if (state is LoginErrorState) {
+            ScaffoldMessenger.of(context)
+                .showSnackBar(const SnackBar(content: Text('Login Error')));
+          }
+        }, builder: (context, state) {
+          var cubit = LoginCubit.get(context);
+          return state.maybeWhen(
+            orElse: () {
+              return SingleChildScrollView(
+                child: Form(
+                  key: formKey,
+                  child: Center(
+                    child: Padding(
+                      padding: const EdgeInsets.all(20),
+                      child: Column(
+                        mainAxisSize: MainAxisSize.min,
+                        children: [
+                          const SizedBox(
+                            height: 70,
+                          ),
+                          Text(
+                            login,
+                            style: Theme.of(context).textTheme.headline6,
+                          ),
+                          const SizedBox(
+                            height: 20,
+                          ),
+                          CustomTextFormField(
+                            controller: emailController,
+                            icon: Icons.email_outlined,
+                            inputType: TextInputType.emailAddress,
+                            hint: email,
+                            validate: (value) {
+                              if (value != null) {
+                                bool validation = emailRegExp.hasMatch(value);
+                                if (!validation) return 'Invalid Email';
+                              }
+                              return null;
+                            },
+                          ),
+                          const SizedBox(
+                            height: 20,
+                          ),
+                          CustomTextFormField(
+                            controller: passwordController,
+                            icon: Icons.lock,
+                            inputType: TextInputType.visiblePassword,
+                            hint: password,
+                            obscureText: !_isPasswordShown,
+                            suffixIcon: _isPasswordShown
+                                ? Icons.visibility_outlined
+                                : Icons.visibility_off_outlined,
+                            onSuffixIconPressed: () {
+                              _isPasswordShown = !_isPasswordShown;
+                              setState(() {});
```

Comment:
Isolate widgets that will be rebuilt from their parent so the whole screen doesn't get rebuilt redundantly.